include/extractor/guidance/toolkit.hpp

```diff
@@ -350,8 +350,9 @@ inline bool requiresNameAnnounced(const std::string &from, const std::string &to
         const auto ref_begin = name.find_first_of('(');
         if (ref_begin != std::string::npos)
         {
-            out_name = name.substr(0, ref_begin);
-            out_ref = name.substr(ref_begin + 1, name.find_first_of(')') - 1);
+            out_name = name.substr(0, ref_begin-1);
```

Comment:
it should theoretically work, based on the concatenation in the form of `+ "_("`. To be on the save side, I have changed it to check for ref_begin > 0.